EIPS/eip-2537.md

```diff
@@ -24,28 +24,30 @@ The motivation of this precompile is to add a cryptographic primitive that allow
-| BLS12_MAP_FP_TO_G1  | 0x12  | precompile address |
-| BLS12_MAP_FP2_TO_G2 | 0x13  | precompile address |
-
-If `block.timestamp >= FORK_TIMESTAMP` we introduce *nine* separate precompiles to perform the following operations:
+| Name                       | Value | Comment            |
+|----------------------------|-------|--------------------|
+| `FORK_TIMESTAMP`           | *TBD* | Mainnet            |
+| BLS12_G1ADD                | 0x0b  | precompile address |
+| BLS12_G1MSM                | 0x0c  | precompile address |
+| BLS12_G1MSM_UNSAFE         | 0x0d  | precompile address |
+| BLS12_G2ADD                | 0x0e  | precompile address |
+| BLS12_G2MSM                | 0x0f  | precompile address |
+| BLS12_G2MSM_UNSAFE         | 0x10  | precompile address |
+| BLS12_PAIRING_CHECK        | 0x11  | precompile address |
+| BLS12_PAIRING_CHECK_UNSAFE | 0x12  | precompile address |
+| BLS12_MAP_FP_TO_G1         | 0x13  | precompile address |
+| BLS12_MAP_FP2_TO_G2        | 0x14  | precompile address |
+
+If `block.timestamp >= FORK_TIMESTAMP` we introduce *ten* separate precompiles to perform the following operations:
 
 - BLS12_G1ADD - to perform point addition in G1 (curve over base prime field) with a gas cost of `500` gas
-- BLS12_G1MUL - to perform point multiplication in G1 (curve over base prime field) with a gas cost of `12000` gas
-- BLS12_G1MSM - to perform multi-scalar-multiplication (MSM) in G1 (curve over base prime field) with a gas cost formula defined in the corresponding section
+- BLS12_G1MSM - to perform multi-scalar-multiplication (MSM) in G1 (curve over base prime field) with a gas cost formula defined in the corresponding section. In the base case of one point this will be a simple point multiplication.
+- BLS12_G1MSM_UNSAFE - identical to BLS12_G1MSM except that subgroup checks will **not** be performed.
 - BLS12_G2ADD - to perform point addition in G2 (curve over quadratic extension of the base prime field) with a gas cost of `800` gas
-- BLS12_G2MUL - to perform point multiplication in G2 (curve over quadratic extension of the base prime field) with a gas cost of `45000` gas
-- BLS12_G2MSM - to perform multi-scalar-multiplication (MSM) in G2 (curve over quadratic extension of the base prime field) with a gas cost formula defined in the corresponding section
+- BLS12_G2MSM - to perform multi-scalar-multiplication (MSM) in G2 (curve over quadratic extension of the base prime field) with a gas cost formula defined in the corresponding section. In the base case of one point this will be a simple point multiplication.
+- BLS12_G2MSM_UNSAFE - identical to BLS12_G2MSM except that subgroup checks will **not** be performed.
 - BLS12_PAIRING_CHECK - to perform a pairing operations between a set of *pairs* of (G1, G2) points a gas cost formula defined in the corresponding section
+- BLS12_PAIRING_CHECK_UNSAFE - identical to BLS12_PAIRING_CHECK except that subgroup checks will **not** be performed.
```

Comment:
I think this is no-starter. How are you going to define this operation for points outside of the main prime subgroup?

Reply:
totally agree with @chfast  here!!